components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -163,4 +179,34 @@ mod tests {
             assert_eq!(output, None, "{:?} {:?}", date, format);
         }
     }
+
+    #[test]
+    fn test_week_day() {
+        let cases = vec![
+            ("2018-12-03", Some(0i64)),
+            ("2018-12-04", Some(1i64)),
+            ("2018-12-05", Some(2i64)),
+            ("2018-12-06", Some(3i64)),
+            ("2018-12-07", Some(4i64)),
+            ("2018-12-08", Some(5i64)),
+            ("2018-12-09", Some(6i64)),
+            ("0000-00-00", None),
+            ("2018-12-00", None),
+            ("2018-00-03", None),
+        ];
+        let mut ctx = EvalContext::default();
+        for (arg, exp) in cases {
+            let datetime = Some(DateTime::parse_datetime(&mut ctx, arg, 6, true).unwrap());
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(datetime.clone())
+                .evaluate(ScalarFuncSig::WeekDay)
+                .unwrap();
+            assert_eq!(output, exp);
+        }
+        let output = RpnFnScalarEvaluator::new()
```

Comment:
How about using `test_unary_func_ok_none` to simplify your code？

Reply:
Rest LGTM.

Reply:
@mapleFU Well `test_unary_func_ok_none` cannot be used here since `crate::rpn_expr::impl_math::tests` is defined as private. Maybe this function should be moved to another public module instead of a test module.